meteor/server/api/rundownLayouts.ts

```diff
@@ -116,6 +141,26 @@ PickerGET.route('/shelfLayouts/download/:id', async (params, _req: IncomingMessa
 	res.end(content)
 })
 
+function validateAndGetRestoredLayoutBody(req: IncomingMessage): RundownLayoutBase {
+	if (!req.body) throw new Meteor.Error(400, 'Restore Shelf Layout: Missing request body')
+
+	if (typeof req.body !== 'object') throw new Meteor.Error(400, 'Restore Shelf Layout: Invalid request body')
+
+	const layout: Partial<RundownLayoutBase> = req.body
+
+	check(layout._id, Match.Optional(String))
+	check(layout.name, String)
+	check(layout.type, String)
+
+	return layout as RundownLayoutBase
+}
+
+async function upsertLayout(layout: RundownLayoutBase, showStyleBase: ShowStyleBaseLight) {
```

Comment:
Don't know how strict we are with function return types, but this one is implicitly returning `Promise<void>`.

Also what does `upsert` mean?
Is it inserting something into e.g a parent?
Name is very likely fine, i am just missing the knowledge needed to know what the means.

Reply:
https://en.wiktionary.org/wiki/upsert

Reply:
> Don't know how strict we are with function return types,

I'm as strict as the linting rules are. If something so simple can be inferred, I will allow it.